dashboards-notebooks/public/components/paragraph_components/para_query_grid.tsx

```diff
@@ -71,18 +81,44 @@ export function QueryDataGrid(props: QueryDataGridProps) {
   );
 
   const renderCellValue = useMemo(() => {
-    return ({ rowIndex, columnId }) => {
+    return ({ rowIndex, columnId }: RenderCellValueProps) => {
       return dataValues.hasOwnProperty(rowIndex)
         ? dataValues[rowIndex][columnId]
         : null;
     };
   }, []);
 
+  const getUpdatedVisibleColumns = (queryColumns: Array<Object>) => {
+    let updatedVisibleColumns = [];
+    for (let index = 0; index < queryColumns.length; ++index) {
+      updatedVisibleColumns.push(queryColumns[index].displayAsText);
+    }
+    return updatedVisibleColumns;
+  }
+
+  
   useEffect(() => {
-  }, [visibleColumns, sortingColumns]);
+    if ($('.euiDataGrid__overflow').is(':visible')) {
+      setIsVisible(true);
+    }
+    setTimeout(() => {
+      if ($('.euiDataGrid__overflow').is(':visible')) {
+        setIsVisible(true);
+      }
+    }, 1000);
```

Comment:
why is `if ($('.euiDataGrid__overflow').is(':visible'))` needed? seems it should be visible after `dataValues` gets passed in?

Reply:
This is based on UX feedback to add a loading spinner until the data grid loads